### PR TITLE
Fix javascript hashing bug

### PR DESCRIPTION
**Pull request by ericvergnaud**
Completely fixes bug found by @akaJes <- thanks
@parrt fully tested, good for merge

### CODE REVIEW

File: contributors.txt [hunk 1]
```diff
@@ -221,3 +221,4 @@ YYYY/MM/DD, github id, Full name, email
 2019/07/11, olowo726, Olof Wolgast, olof@baah.se
 2019/07/16, abhijithneilabraham, Abhijith Neil Abraham, abhijithneilabrahampk@gmail.com
 2019/07/26, Braavos96, Eric Hettiaratchi, erichettiaratchi@gmail.com
+2019/08/23, akaJes, Oleksandr Mamchyts, akaJes@gmail.com
```
Comment (sharwell) [inline contributors.txt]: ❗️ Contributors are not allowed to sign the agreement for a different contributor. @akaJes will need to submit this commit directly for it to be acceptable.

Comment (ericvergnaud) [reply contributors.txt]: He has already done so, see #2632, I pre-merged it for convenience

**Comment by ericvergnaud** [comment]
@parrt can you merge this one?

**Comment by parrt** [comment]
thanks, gang!

**Pull request merged**